### PR TITLE
fix: Export as PDF button is disabled when chat is closed by the visitor

### PR DESCRIPTION
**Pull request by filipemarins**
## Proposed changes (including videos or screenshots)

Fix the `export as PDF` button is disabled when chat is closed by the visitor;

## Issue(s)

[OC-673](https://rocketchat.atlassian.net/browse/OC-673) [OC-725](https://rocketchat.atlassian.net/browse/OC-725) [OC-695](https://rocketchat.atlassian.net/browse/OC-695)

## Further comments

Move the room data subscription stream to the scope of the room provider. This is necessary as we require this property to be updated on other components.

[OC-673]: https://rocketchat.atlassian.net/browse/OC-673?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aleksandernsilva — DISMISSED** [review]

File: apps/meteor/client/views/room/providers/RoomProvider.tsx [hunk 1]
```diff
@@ -27,6 +28,22 @@ const RoomProvider = ({ rid, children }: RoomProviderProps): ReactElement => {
 
 	const roomQuery = useReactiveQuery(['rooms', rid], ({ rooms }) => rooms.findOne({ _id: rid }));
 
+	const subscribeToRoom = useStream('room-data');
+
+	const [subscribedData, setSubscribedData] = useState<Pick<IOmnichannelRoom, 'open' | 'servedBy' | 'queuedAt'> | null>(null);
+
+	useEffect(() => {
+		setSubscribedData(null);
```
Comment (ggazzo) [inline apps/meteor/client/views/room/providers/RoomProvider.tsx:36]: why not use `setQueriesData` ?

**Review by ggazzo — CHANGES_REQUESTED** [review]
please check the comments

File: apps/meteor/client/views/room/providers/RoomProvider.tsx [hunk 2]
```diff
@@ -27,6 +28,16 @@ const RoomProvider = ({ rid, children }: RoomProviderProps): ReactElement => {
 
 	const roomQuery = useReactiveQuery(['rooms', rid], ({ rooms }) => rooms.findOne({ _id: rid }));
 
+	const subscribeToRoom = useStream('room-data');
```
Comment (MartinSchoeler) [inline apps/meteor/client/views/room/providers/RoomProvider.tsx:31]: Let's move this to the OmnichannelProvider instead

**Review by MartinSchoeler — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/room/providers/RoomProvider.tsx (hunk 2 above)
Comment (filipemarins) [reply apps/meteor/client/views/room/providers/RoomProvider.tsx:32]: Unfortunately we don't have room context there :/

File: apps/meteor/client/views/room/providers/RoomProvider.tsx [hunk 3]
```diff
@@ -27,6 +29,21 @@ const RoomProvider = ({ rid, children }: RoomProviderProps): ReactElement => {
...
+
+	const queryClient = useQueryClient();
+
+	// TODO: move this to omnichannel context only
+	useEffect(() => {
+		if (!isOmnichannelRoom(room)) {
```
Comment (aleksandernsilva) [inline apps/meteor/client/views/room/providers/RoomProvider.tsx:38]: Just noticed after the last change, `room` doesn't exist anymore at this point, which will crash the app. I'll adjust the code.

**Review by aleksandernsilva — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/room/providers/RoomProvider.tsx (hunk 3 above)
Comment (tassoevan) [reply apps/meteor/client/views/room/providers/RoomProvider.tsx:38]: I've to cautiously use suggestions... I usually just make a draft of an idea, not really ensuring if the code work 🙈

**Review by aleksandernsilva — DISMISSED** [review]

**Review by tassoevan — DISMISSED** [review]

**Review by MartinSchoeler — APPROVED** [review]

**Review by MartinSchoeler — DISMISSED** [review]

**Review by tassoevan — APPROVED** [review]

**Review by MartinSchoeler — APPROVED** [review]

**Review by tassoevan — APPROVED** [review]

**Pull request merged**